server/src/main/java/io/crate/metadata/ColumnIdent.java

```diff
@@ -248,8 +247,8 @@ private static void validateDotInColumnName(String columnName) {
      * @param columnName column name to check for validity
      */
     private static void validateSubscriptPatternInColumnName(String columnName) {
-        if (SUBSCRIPT_PATTERN.matcher(columnName).matches()) {
-            throw new InvalidColumnNameException(columnName, "conflicts with subscript pattern");
+        if (columnName.contains("[")) {
```

Comment:
imho, I think we could relax it, and detect (just a simple example, maybe need to check that works for nested expressions, like `abda[ fdsfda [fdafda [fdafda ]` ): `.*\\[.*\\].*`, so to find a valid opening/closing subscript pattern. what do you think?

Reply:
Thanks for quick reviewing, could you clarify the purpose for that? I find it similar to the original `SUBSCRIPT_PATTERN.matcher(columnName).matches()`.

Reply:
It's just a thought, since we care only to prevent valid subscript expressions, but on a second thought, it's fine to forbid it completely.

Reply:
My thinking is more of tools that could generate some "crazy" strings for column names, and since we only care about a valid subscript expression (to forbid), I was thinking if we maybe want to detect just this, and not in general the `[`.

But I'd say let's keep it as is and we can always revisit if we stumble upon such a requirement.